packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -190,11 +201,232 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const searchInputRef = useRef<HTMLInputElement>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const [isMineOnly, setIsMineOnly] = useState(false);
+    const [searchText, setSearchText] = useState('');
+    const closeQueryImporter = (): void => {
+      editorStore.queryLoaderState.setIsQueryLoaderOpen(false);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const toggleIsMineOnly = (): void => {
+      setIsMineOnly(!isMineOnly);
+    };
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.queryLoaderState.setIsQueryLoaderOpen(false);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
+        applicationStore.setBlockingAlert({
+          message: 'Loading query...',
+          showLoading: true,
+        });
+      } else {
```

Comment:
I think we don't need this else block, instead, let's change the tooltip of the `Load Query` button when it's disabled because user hasn't selected any query.